src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer producer = getProducer(type);
+        if (producer == null) {
+            return new ArrayList<>();
+        }
+        List<? extends Machine> retVal = producer.get();
+        return new ArrayList<>(retVal);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        for (Machine machine : machines) {
+            machine.doWork();
+        }
+    }
+
+    private MachineProducer getProducer(Class type) {
+        if (Bulldozer.class.equals(type)) {
+            return new BulldozerProducer();
+        }
+        if (Excavator.class.equals(type)) {
+            return new ExcavatorProducer();
+        }
+        if (Truck.class.equals(type)) {
+            return new TruckProducer();
+        }
+        return null;
```

Comment:
It's bad practice to return null, lets throw some exception instead

Reply:
I agree, moreover initially this function thrown IllegalArgumentException. But, one test (getAll_invalidType()) faulted. So, i checked [https://mate-academy.github.io/jv-program-common-mistakes/java-core/pecs/pecs.html] and find out there : 

> Don’t return null or specific type List in getAll() when no producer has been found.
Returning null is considered to be a bad practice: a user of your method needs to write additional null checks when using it. In other cases NPE might appear. Returning List of a specific type, like List<Truck> when no producer found is a bad practice as well, because it’s not an obvious behaviour, instead it’s better to return empty List.

That's why method returns empty List

Reply:
Agree, lets return empty list